components/txn_types/src/write.rs

```diff
@@ -409,6 +437,9 @@ impl WriteRef<'_> {
         if !self.last_change_ts.is_zero() {
             size += 1 + size_of::<u64>() + MAX_VAR_U64_LEN;
         }
+        if self.txn_source != 0 {
```

Comment:
It looks better if you write code consistently. Use `!= 0` or `> 0` in both lock and write..